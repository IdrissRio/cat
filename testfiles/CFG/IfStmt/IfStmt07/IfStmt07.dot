digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[22]"; "Opt[25]"; "List[26]"; "List[27]"; "InstanceInitialization[6]"; "InstanceInitialization[14]"}
{ rank = same; "List[23]"; "InstanceInitializer[28]"; "Entry[7]"; "Exit[8]"; "Entry[15]"; "Exit[16]"}
{ rank = same; "Modifier[24]"; "Block[29]"}
{ rank = same; "List[30]"}
{ rank = same; "IfStmt[31]"; "VarDeclStmt[67]"}
{ rank = same; "BooleanLiteral[32]"; "Block[33]"; "Opt[43]"; "Modifiers[68]"; "PrimitiveTypeAccess[70]"; "List[71]"}
{ rank = same; "List[34]"; "IfStmt[44]"; "List[69]"; "VariableDeclarator[72]"}
{ rank = same; "VarDeclStmt[35]"; "BooleanLiteral[45]"; "Block[46]"; "Opt[56]"; "List[73]"; "Opt[74]"}
{ rank = same; "Modifiers[36]"; "PrimitiveTypeAccess[38]"; "List[39]"; "List[47]"; "Block[57]"}
{ rank = same; "List[37]"; "VariableDeclarator[40]"; "VarDeclStmt[48]"; "List[58]"}
{ rank = same; "List[41]"; "Opt[42]"; "Modifiers[49]"; "PrimitiveTypeAccess[51]"; "List[52]"; "VarDeclStmt[59]"}
{ rank = same; "List[54]"; "Opt[55]"; "List[61]"; "VariableDeclarator[64]"}
{ rank = same; "List[50]"; "VariableDeclarator[53]"; "Modifiers[60]"; "PrimitiveTypeAccess[62]"; "List[63]"}
{ rank = same; "List[65]"; "Opt[66]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < IfStmt07 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[22]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[23]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[24]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[25]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[27]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"InstanceInitializer[28]" [label="InstanceInitializer\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[29]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[30]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IfStmt[31]" [label="IfStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"BooleanLiteral[32]" [label="BooleanLiteral\n < true >", fillcolor=white   style=filled  ]
"Block[33]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[34]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[35]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[36]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[37]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[38]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[39]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[40]" [label="VariableDeclarator\n < a >", fillcolor=white   style=filled  ]
"List[41]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[42]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[43]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IfStmt[44]" [label="IfStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"BooleanLiteral[45]" [label="BooleanLiteral\n < true >", fillcolor=white   style=filled  ]
"Block[46]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[47]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[48]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[49]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[50]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[51]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[52]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[53]" [label="VariableDeclarator\n < b >", fillcolor=white   style=filled  ]
"List[54]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[55]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[56]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[57]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[58]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[59]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[60]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[61]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[62]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[63]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[64]" [label="VariableDeclarator\n < c >", fillcolor=white   style=filled  ]
"List[65]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[66]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[67]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[68]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[69]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[70]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[71]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[72]" [label="VariableDeclarator\n < d >", fillcolor=white   style=filled  ]
"List[73]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[74]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"InstanceInitialization[6]" [label="InstanceInitialization\n < Instance >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[7]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[8]" [label="Exit\n ", fillcolor=white   style=filled  ]
"InstanceInitialization[14]" [label="InstanceInitialization\n < Static >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[15]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[16]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[23]" -> "Modifier[24]" [style=dashed, color=gray]
"Modifiers[22]" -> "List[23]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[22]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[25]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[26]" [style=dashed, color=gray]
"IfStmt[31]" -> "BooleanLiteral[32]" [style=dashed, color=gray]
"Modifiers[36]" -> "List[37]" [style=dashed, color=gray]
"VarDeclStmt[35]" -> "Modifiers[36]" [style=dashed, color=gray]
"VarDeclStmt[35]" -> "PrimitiveTypeAccess[38]" [style=dashed, color=gray]
"VariableDeclarator[40]" -> "List[41]" [style=dashed, color=gray]
"VariableDeclarator[40]" -> "Opt[42]" [style=dashed, color=gray]
"List[39]" -> "VariableDeclarator[40]" [style=dashed, color=gray]
"VarDeclStmt[35]" -> "List[39]" [style=dashed, color=gray]
"List[34]" -> "VarDeclStmt[35]" [style=dashed, color=gray]
"Block[33]" -> "List[34]" [style=dashed, color=gray]
"IfStmt[31]" -> "Block[33]" [style=dashed, color=gray]
"IfStmt[44]" -> "BooleanLiteral[45]" [style=dashed, color=gray]
"Modifiers[49]" -> "List[50]" [style=dashed, color=gray]
"VarDeclStmt[48]" -> "Modifiers[49]" [style=dashed, color=gray]
"VarDeclStmt[48]" -> "PrimitiveTypeAccess[51]" [style=dashed, color=gray]
"VariableDeclarator[53]" -> "List[54]" [style=dashed, color=gray]
"VariableDeclarator[53]" -> "Opt[55]" [style=dashed, color=gray]
"List[52]" -> "VariableDeclarator[53]" [style=dashed, color=gray]
"VarDeclStmt[48]" -> "List[52]" [style=dashed, color=gray]
"List[47]" -> "VarDeclStmt[48]" [style=dashed, color=gray]
"Block[46]" -> "List[47]" [style=dashed, color=gray]
"IfStmt[44]" -> "Block[46]" [style=dashed, color=gray]
"Modifiers[60]" -> "List[61]" [style=dashed, color=gray]
"VarDeclStmt[59]" -> "Modifiers[60]" [style=dashed, color=gray]
"VarDeclStmt[59]" -> "PrimitiveTypeAccess[62]" [style=dashed, color=gray]
"VariableDeclarator[64]" -> "List[65]" [style=dashed, color=gray]
"VariableDeclarator[64]" -> "Opt[66]" [style=dashed, color=gray]
"List[63]" -> "VariableDeclarator[64]" [style=dashed, color=gray]
"VarDeclStmt[59]" -> "List[63]" [style=dashed, color=gray]
"List[58]" -> "VarDeclStmt[59]" [style=dashed, color=gray]
"Block[57]" -> "List[58]" [style=dashed, color=gray]
"Opt[56]" -> "Block[57]" [style=dashed, color=gray]
"IfStmt[44]" -> "Opt[56]" [style=dashed, color=gray]
"Opt[43]" -> "IfStmt[44]" [style=dashed, color=gray]
"IfStmt[31]" -> "Opt[43]" [style=dashed, color=gray]
"List[30]" -> "IfStmt[31]" [style=dashed, color=gray]
"Modifiers[68]" -> "List[69]" [style=dashed, color=gray]
"VarDeclStmt[67]" -> "Modifiers[68]" [style=dashed, color=gray]
"VarDeclStmt[67]" -> "PrimitiveTypeAccess[70]" [style=dashed, color=gray]
"VariableDeclarator[72]" -> "List[73]" [style=dashed, color=gray]
"VariableDeclarator[72]" -> "Opt[74]" [style=dashed, color=gray]
"List[71]" -> "VariableDeclarator[72]" [style=dashed, color=gray]
"VarDeclStmt[67]" -> "List[71]" [style=dashed, color=gray]
"List[30]" -> "VarDeclStmt[67]" [style=dashed, color=gray]
"Block[29]" -> "List[30]" [style=dashed, color=gray]
"InstanceInitializer[28]" -> "Block[29]" [style=dashed, color=gray]
"List[27]" -> "InstanceInitializer[28]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[27]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Entry[7]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Exit[8]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Entry[15]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Exit[16]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[14]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[16]" -> "Entry[15]" [color=red, constraint=false]
"Entry[15]" -> "Exit[16]" [color=blue, constraint=false, xlabel=" "] 
"Exit[8]" -> "VariableDeclarator[72]" [color=red, constraint=false]
"VariableDeclarator[72]" -> "Exit[8]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[72]" -> "VariableDeclarator[40]" [color=red, constraint=false]
"VariableDeclarator[72]" -> "VariableDeclarator[53]" [color=red, constraint=false]
"VariableDeclarator[72]" -> "VariableDeclarator[64]" [color=red, constraint=false]
"VariableDeclarator[40]" -> "VariableDeclarator[72]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[40]" -> "BooleanLiteral[32]" [color=red, constraint=false]
"VariableDeclarator[53]" -> "VariableDeclarator[72]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[53]" -> "BooleanLiteral[45]" [color=red, constraint=false]
"VariableDeclarator[64]" -> "VariableDeclarator[72]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[64]" -> "BooleanLiteral[45]" [color=red, constraint=false]
"BooleanLiteral[45]" -> "VariableDeclarator[53]" [color=blue, constraint=false, xlabel="T "] 
"BooleanLiteral[45]" -> "VariableDeclarator[64]" [color=blue, constraint=false, xlabel="F "] 
"BooleanLiteral[45]" -> "BooleanLiteral[32]" [color=red, constraint=false]
"BooleanLiteral[32]" -> "VariableDeclarator[40]" [color=blue, constraint=false, xlabel="T "] 
"BooleanLiteral[32]" -> "BooleanLiteral[45]" [color=blue, constraint=false, xlabel="F "] 
"BooleanLiteral[32]" -> "Entry[7]" [color=red, constraint=false]
"Entry[7]" -> "BooleanLiteral[32]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"InstanceInitialization[14]" -> "InstanceInitialization[6]" -> "Modifiers[22]" -> "Opt[25]" -> "List[26]" -> "List[27]" [style=invis]
"IfStmt[31]" -> "VarDeclStmt[67]" [style=invis]
"BooleanLiteral[32]" -> "Block[33]" -> "Opt[43]" [style=invis]
"Modifiers[36]" -> "PrimitiveTypeAccess[38]" -> "List[39]" [style=invis]
"List[41]" -> "Opt[42]" [style=invis]
"BooleanLiteral[45]" -> "Block[46]" -> "Opt[56]" [style=invis]
"Modifiers[49]" -> "PrimitiveTypeAccess[51]" -> "List[52]" [style=invis]
"List[54]" -> "Opt[55]" [style=invis]
"Modifiers[60]" -> "PrimitiveTypeAccess[62]" -> "List[63]" [style=invis]
"List[65]" -> "Opt[66]" [style=invis]
"Modifiers[68]" -> "PrimitiveTypeAccess[70]" -> "List[71]" [style=invis]
"List[73]" -> "Opt[74]" [style=invis]
"Entry[7]" -> "Exit[8]" [style=invis]
"Entry[15]" -> "Exit[16]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
65", fillcolor=1]
Edges [label="Total Nr. Edges
84", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
9", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
10", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}