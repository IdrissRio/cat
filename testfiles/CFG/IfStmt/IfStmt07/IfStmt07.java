public class IfStmt07 {
  {
    if (true) {
      int a;
    } else if (true) {
      int b;
    } else {
      int c;
    }
    int d;
  }
}