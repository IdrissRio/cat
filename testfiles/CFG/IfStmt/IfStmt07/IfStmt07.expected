NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [BooleanLiteral[32]]
NodeName: BooleanLiteral[32] Successors: [VariableDeclarator[40], BooleanLiteral[45]]
NodeName: VariableDeclarator[40] Successors: [VariableDeclarator[72]]
NodeName: VariableDeclarator[72] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: BooleanLiteral[45] Successors: [VariableDeclarator[53], VariableDeclarator[64]]
NodeName: VariableDeclarator[53] Successors: [VariableDeclarator[72]]
NodeName: VariableDeclarator[64] Successors: [VariableDeclarator[72]]

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [VariableDeclarator[72]]
NodeName: VariableDeclarator[72] Predecessor: [VariableDeclarator[40], VariableDeclarator[53], VariableDeclarator[64]]
NodeName: VariableDeclarator[40] Predecessor: [BooleanLiteral[32]]
NodeName: BooleanLiteral[32] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: VariableDeclarator[53] Predecessor: [BooleanLiteral[45]]
NodeName: BooleanLiteral[45] Predecessor: [BooleanLiteral[32]]
NodeName: VariableDeclarator[64] Predecessor: [BooleanLiteral[45]]
